### PR TITLE
Remove distributed functionality

### PR DESCRIPTION
**Pull request by msimberg**
It looks like #1 was closed due to master being force-pushed with no way to open the PR again. Opening a new PR until I know if I can actually reopen #1 somehow.

### CODE REVIEW

**Comment by msimberg** [comment]
@hkaiser are you planning to have a closer look at this? Or do you have further responses to the comments that @aurianer has made?

**Comment by msimberg** [comment]
launch perftests

File: .circleci/config.yml [hunk 1]
```diff
@@ -192,41 +175,33 @@ jobs:
...
-                -DHPX_WITH_STACKTRACES_DEMANGLE_SYMBOLS=Off \
-                -DHPX_WITH_TESTS_DEBUG_LOG=On \
-                -DHPX_WITH_TESTS_DEBUG_LOG_DESTINATION=/hpx/build/debug-log.txt \
-                -DHPX_WITH_SPINLOCK_DEADLOCK_DETECTION=On \
-                -DHPX_WITH_CHECK_MODULE_DEPENDENCIES=On \
-                -DCMAKE_EXPORT_COMPILE_COMMANDS=On \
-                -DHPX_WITH_DOCUMENTATION=On \
-                -DHPX_WITH_DOCUMENTATION_OUTPUT_FORMATS="${DOCUMENTATION_OUTPUT_FORMATS}"
+                -DCMAKE_CXX_FLAGS="-fcolor-diagnostics" \
+                -DHPXLocal_WITH_GIT_COMMIT=${CIRCLE_SHA1} \
```
[the remark this answers is not in the document]
Comment (msimberg) [reply .circleci/config.yml:184]: I'll leave this as it is for now. Please comment if you think it should be changed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -213,536 +211,389 @@ hpx_option(
...
+  add_library(hpx_local_precompiled_headers OBJECT libs/src/dummy.cpp)
+  add_executable(hpx_local_exe_precompiled_headers libs/src/dummy.cpp)
 
-    if(HPX_WITH_STATIC_EXE_LINKING)
-      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static")
-    endif()
-  endif()
-else()
-  set(hpx_library_link_mode SHARED)
+  set_target_properties(hpx_local_precompiled_headers PROPERTIES FOLDER "Core")
```
[the remark this answers is not in the document]
Comment (msimberg) [reply CMakeLists.txt:275]: This is related to yesterday's discussion about how this is going to show up in Visual Studio. @hkaiser could you report back here once you've tried it?

File: CMakeLists.txt [hunk 3]
```diff
@@ -781,385 +622,254 @@ if(HPX_WITH_STACKTRACES OR HPX_WITH_THREAD_BACKTRACE_ON_SUSPENSION)
...
-    HPX_WITH_ASYNC_MPI
-    BOOL
-    "Enable support for returning futures from MPI asynchronous calls (default: ON if networking+MPI already enabled)"
-    OFF
-    CATEGORY "MPI"
-  )
-endif(HPX_WITH_NETWORKING)
-
-if((HPX_WITH_NETWORKING AND HPX_WITH_PARCELPORT_MPI) OR HPX_WITH_ASYNC_MPI)
+if(HPXLocal_WITH_ASYNC_MPI)
```
[the remark this answers is not in the document]
Comment (msimberg) [reply CMakeLists.txt:862]: Based on yesterday's discussion that we'll keep the MPI functionality here for now, but move it out later, I'll keep the name. Once it moves out it'll be completely removed anyway.

**Comment by msimberg** [comment]
@hkaiser I've added a check that prints an error message as follows if it detects inconsistently cased CMake cache variables:
```
CMake Error at cmake/HPXLocal_Message.cmake:51 (message):
  HPXLocal expects all HPXLocal variables to be prefixed with HPXLocal.
  Found the following variables with inconsistent casing: HPXLOCAL_WITH_BLAH
  HPX_LOCAL_WITH_BOO.  Please unset the listed variables and set them with
  the correctly cased prefix instead.
Call Stack (most recent call first):
  cmake/HPXLocal_CheckOptionsCase.cmake:20 (hpx_local_error)
  CMakeLists.txt:113 (include)


-- Configuring incomplete, errors occurred!
```
Let me know if you think a different message would be better. The check takes a measurable but negligible amount of time (a few tens of ms on my system), so I think we're ok here.

File: cmake/installed_hpx.cmake [hunk 4]
```diff
@@ -7,119 +7,126 @@
 find_package(HPX REQUIRED)
 
 # ##############################################################################
-hpx_option(
-  HPX_WITH_EXAMPLES BOOL "Build the HPX examples (default OFF)" OFF
+hpx_local_option(
+  HPXLocal_WITH_EXAMPLES BOOL "Build the HPX examples (default OFF)" OFF
```
[the remark this answers is not in the document]
Comment (hkaiser) [reply cmake/installed_hpx.cmake:11]: @msimberg I don't know anybody who is using this functionality.

**Comment by hkaiser** [comment]
@msimberg thanks for moving this forward!

**Pull request merged**